added up auth module.

abheektripathy opened the pull request:
added up auth module, verified the endpoints plus initialised the next.js frontend

ManofWax commented on buffalogs/authentication/migrations/0001_initial.py:
```diff
@@ -0,0 +1,82 @@
+                    "username",
+                    models.CharField(db_index=True, max_length=255, unique=True),
+                ),
+                (
+                    "email",
+                    models.EmailField(db_index=True, max_length=255, unique=True),
+                ),
+                ("is_staff", models.BooleanField(default=False)),
+                ("is_verified", models.BooleanField(default=False)),
+                ("created_at", models.DateTimeField(auto_now_add=True)),
+                ("updated_at", models.DateTimeField(auto_now=True)),
+                ("avatar", models.CharField(default="", max_length=225)),
+                (
+                    "groups",
+                    models.ManyToManyField(
+                        blank=True,
+                        help_text="The groups this user belongs to. A user will get all permissions granted to each of their groups.",
+                        related_name="user_set",
+                        related_query_name="user",
+                        to="auth.group",
+                        verbose_name="groups",
+                    ),
+                ),
+                (
+                    "user_permissions",
+                    models.ManyToManyField(
+                        blank=True,
+                        help_text="Specific permissions for this user.",
+                        related_name="user_set",
+                        related_query_name="user",
+                        to="auth.permission",
+                        verbose_name="user permissions",
+                    ),
+                ),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+    ]
```
Don't use the `User` table from buffalogs app: that user table is reserved f or users detected in logs processed by buffalogs not for the users that use the project.

Please create a new User model on the `authentication` app you've just created. This new model should be used for authetication

ManofWax commented:
Hi, it looks good. Can you lint with black as described here: https://github.com/certego/BuffaLogs/blob/main/CONTRIBUTING.md#pull-request (step 2)

abheektripathy commented:
added up protected routes/setup middleware and fixed auth on the frontend.
<img width="660" alt="Screenshot 2023-06-24 at 1 21 48 PM" src="https://github.com/certego/BuffaLogs/assets/90976669/db860a31-e401-4e26-8cc6-54836315db87">
